File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit=0;
        int n = prices.length;
        int min=prices[0];
        for(int i=0; i<n; i++){
            int temp = prices[i]-min;
            profit=Math.max(profit, temp);
            min= Math.min(min, prices[i]);
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minvalue = prices[0]
        profit = 0
        for num in prices:
            temp = num-minvalue
            profit = max(profit, temp)
            if minvalue > num:
                minvalue = num
        return profit
            
        